server/events/policy_filter.go

```diff
@@ -19,15 +29,26 @@ type teamMemberFetcher interface {
 type ApprovedPolicyFilter struct {
 	owners sync.Map //cache
```

Comment:
remind me how this gets refreshed

Reply:
Any time there is a cache miss when looking for a specific user. https://github.com/lyft/atlantis/blob/d2345e7dfc389b0a5ab91537b37d63ad1a27bd55/server/events/policy_filter.go#L134-L145

Two cases we call for it:

- [Case 1](https://github.com/lyft/atlantis/blob/d2345e7dfc389b0a5ab91537b37d63ad1a27bd55/server/events/policy_filter.go#L76): When checking to see if the PR has an approval from a failed policy owner, thus we can filter out the failed policy
- [Case 2](https://github.com/lyft/atlantis/blob/d2345e7dfc389b0a5ab91537b37d63ad1a27bd55/server/events/policy_filter.go#L123): When first looking at all of the existing PR approvals to identify if any belong to an owner and can be dismissed

Reply:
> Any time there is a cache miss when looking for a specific user.

Is this right? What if someone gets removed from the owners list?  

If you're caching, typical practice is to to some time based expiry.  Managing deltas gets tricky and is somewhat error prone imo.

Reply:
Discussed offline, I removed cache entirely.